### PR TITLE
Feeds api improvements

### PR DESCRIPTION
**Pull request by rybmat**
* cache image data for article information
* call user-attribute service only once when collecting information about top editors

@Wikia/iwing

### CODE REVIEW

File: extensions/wikia/FeedsAndPosts/ArticleData.class.php [hunk 1]
```diff
@@ -5,19 +5,26 @@
...
 use ImageServing;
 use Title;
+use WikiaDataAccess;
 
 class ArticleData {
-	public static function getImages(int $articleId, $limit=11): array {
-		$imageData = ( new ImageServing( [ $articleId ] ) )->getImages( $limit )[strval( $articleId )] ?? [];
+	public static function getImages( int $articleId, $limit = 11 ): array {
+		return WikiaDataAccess::cache(
+			wfMemcKey('feeds', 'article-images'),
```
Comment (mszabo-wikia) [inline extensions/wikia/FeedsAndPosts/ArticleData.class.php]: This needs to vary on article ID.

File: extensions/wikia/FeedsAndPosts/ArticleData.class.php [hunk 2]
```diff
@@ -5,19 +5,26 @@
...
+			10800, // 3h
+			function () use ( $articleId, $limit ) {
+				$imageData = ( new ImageServing( [ $articleId ] ) )->getImages( $limit )[strval( $articleId )] ?? [];
```
Comment (mszabo-wikia) [inline extensions/wikia/FeedsAndPosts/ArticleData.class.php]: Let's split this into multiple lines, it's a bit difficult to follow.

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/FeedsAndPosts/ArticleData.class.php (hunk 1 above)
Comment (rybmat) [reply extensions/wikia/FeedsAndPosts/ArticleData.class.php]: Totally, how could I miss that...

File: extensions/wikia/FeedsAndPosts/ArticleData.class.php (hunk 2 above)
Comment (rybmat) [reply extensions/wikia/FeedsAndPosts/ArticleData.class.php]: I tried various ways and all of them did not look better.. Actually this single line version seems to be most readable

Comment (mszabo-wikia) [reply extensions/wikia/FeedsAndPosts/ArticleData.class.php]: I was thinking of something like
```
$imageServing = new ImageServing( [ $articleId ] );
$images = $imageServing->getImages( $limit );

$imageInfo = $images[strval( $articleId )] ?? [];

Comment (rybmat) [reply extensions/wikia/FeedsAndPosts/ArticleData.class.php]: k, makes sense

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**